pkg/machine/proxyenv/env.go

```diff
@@ -0,0 +1,54 @@
+package proxyenv
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/containers/common/libnetwork/etchosts"
+	"github.com/containers/common/pkg/config"
+	"github.com/containers/podman/v5/pkg/machine"
+	"github.com/containers/podman/v5/pkg/machine/vmconfigs"
+	"github.com/sirupsen/logrus"
+)
+
+const proxySetupScriptTemplate = `#!/bin/bash
+
+SYSTEMD_CONF=/etc/systemd/system.conf.d/default-env.conf
+ENVD_CONF=/etc/environment.d/default-env.conf
+PROFILE_CONF=/etc/profile.d/default-env.sh
+
+mkdir -p /etc/profile.d /etc/environment.d /etc/systemd/system.conf.d/
+rm -f $SYSTEMD_CONF $ENVD_CONF $PROFILE_CONF
+
+echo "[Manager]" >> $SYSTEMD_CONF
+for proxy in %s; do
+	printf "DefaultEnvironment=%%q\n" "$proxy"  >> $SYSTEMD_CONF
+	printf "%%q\n" "$proxy"  >> $ENVD_CONF
+	printf "export %%q\n" "$proxy" >> $PROFILE_CONF
+done
+
+systemctl daemon-reload
+`
+
+func getProxyScript() io.Reader {
+	var envs []string
+	for _, key := range config.ProxyEnv {
+		if value, ok := os.LookupEnv(key); ok {
+			// FIXME: Is this correct for WSL? I don't think so.
+			v := strings.ReplaceAll(value, "127.0.0.1", etchosts.HostContainersInternal)
+			v = strings.ReplaceAll(v, "localhost", etchosts.HostContainersInternal)
```

Comment:
@Luap99 I think you can leave this for now, but host.containers.internal has an unusual meaning in WSL that isnt quite right / what people expect, but does have a valid use. The issue is that host.containers.internal points to the Linux machine IP which makes sense, but it's not the Windows IP. While ports opened on Linux localhost get forwarded for Windows apps, ports opened on Windows do not get forwarded for clients running on Linux. The net effect here is that if you are running a proxy on the podman machine WSL distro, this code you have here is great. However if the proxy is running as a Windows process, it will be the wrong address. 

I hope to move to requiring WSL networking mirroring when it's stable/reliable to resolve this complexity which will have some better semantics. I have also thought about introducing another host entry to differentiate between the two addresses as a temporary solution. In either case though I wouldn't worry about solving it now.

Reply:
But `host.containers.internal` will not be resolved at all without gvproxy in the VM. Podman only adds this entry to /etc/hosts for containers so the actual VM will not have the correct proxy url and fail to pull images for example.